Release/v2.0.0

mawiswiss opened the pull request:
Here we go. I went through all the issues we had. Solved them and cleaned up more issues.

Since we're adding PER Coding Style 2.0 (https://www.php-fig.org/per/coding-style/), it's a major release.
I ran it on a local project, and I could auto-fix all changes. The following rules had to be changed:

- PhpCsFixer\Fixer\Operator\ConcatSpaceFixer::class
- PhpCsFixer\Fixer\ControlStructure\TrailingCommaInMultilineFixer::class
- PhpCsFixer\Fixer\FunctionNotation\FunctionDeclarationFixer::class
- PhpCsFixer\Fixer\Basic\SingleLineEmptyBodyFixer::class
- PhpCsFixer\Fixer\Operator\NewWithParenthesesFixer::class
- PhpCsFixer\Fixer\ClassNotation\ClassDefinitionFixer::class

The updates @rugbymauri is working on with CS Fixer aren't ready yet, and I created a milestone for those adjustments. Once this is ready, we see if there's another major needed or if it's backwards compatible. For now, I want to release those adjustments to have a clean base again.

mawiswiss committed:
feat: unify rule sets and cleanup/adjust rules

mawiswiss committed:
docs: add example config

mawiswiss committed:
chore: minor update to composer packages

mawiswiss committed:
docs: add changelog for new v2 release and adjust README

rugbymauri commented on config/whatwedo-common.php:
```diff
@@ -84,23 +89,29 @@
         ScopeIndentSniff::class => null,
         ScopeClosingBraceSniff::class => null,
         CastSpacesFixer::class => null,
-        NoTrailingCommaInListCallFixer::class => null,
```
https://cs.symfony.com/doc/rules/control_structure/trailing_comma_in_multiline.html

mawiswiss committed:
feat: change rules according to Symfony's PER-CS2.0 ruleset

The PER-CS2.0 ruleset of Symfony doesn't match the documentation. We choose the more readable version, which is also part of the official PER-CS2.0 ruleset (https://cs.symfony.com/doc/ruleSets/PER-CS2.0.html)

#15

mawiswiss committed:
chore: remove IDE project files

mawiswiss committed:
docs: add a note for contributors to check the project for the coding standards as well

mawiswiss committed:
style: fix code style in DumpFixer

rugbymauri commented:
the SymfonySet defines

```
'single_line_empty_body' => false, // overrides @PER-CS2.0
```
 https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/RuleSet/Sets/SymfonySet.php#L206


our ECS config enforces it

rugbymauri commented:
Same issuse

```
            'trailing_comma_in_multiline' => [
                'after_heredoc' => true,
                'elements' => [ // explicitly omit 'arguments'
                    'array_destructuring',
                    'arrays',
                    'match',
                    'parameters',
                ],
            ],
```

https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/RuleSet/Sets/SymfonySet.php#L219


our ECS enforces for 'arguments'
